internal/security-governance-system/webapp/src/main/java/org/wso2/security/tools/sgs/webapp/service/ScanService.java

```diff
@@ -0,0 +1,233 @@
+
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.multipart.MultipartFile;
+import org.wso2.security.tools.sgs.webapp.entity.Scan;
+import org.wso2.security.tools.sgs.webapp.entity.Scanner;
+import org.wso2.security.tools.sgs.webapp.handlers.HttpRequestHandler;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.wso2.security.tools.sgs.webapp.config.StartUpInit.sgsWebappConfiguration;
+
+/**
+ * Service layer methods to handle general methods
+ */
+@Service
+public class ScanService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ScanService.class);
+
+    private static final String ID = "id";
+    private static final String NAME = "name";
+    private static final String OWNER = "owner";
+    private static final String STATUS = "status";
+    private static final String SCHEME = "http";
+
+    /**
+     * Get Scanner details for a given scanner ID
+     *
+     * @param scannerID
+     * @return
+     */
+    public Scanner getScanner(String scannerID) {
+
+        for (Scanner scanner : sgsWebappConfiguration.getStaticScanners()) {
+            if (scanner.getId().equals(scannerID)) {
+                return scanner;
+            }
+        }
+
+        for (Scanner scanner : sgsWebappConfiguration.getDynamicScanners()) {
+            if (scanner.getId().equals(scannerID)) {
+                return scanner;
+            }
+        }
+
+        for (Scanner scanner : sgsWebappConfiguration.getDependencyScanners()) {
+            if (scanner.getId().equals(scannerID)) {
+                return scanner;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Start a scan
+     *
+     * @param multipartFileMultiValueMap
+     * @param parameterMap
+     * @return
+     */
+    public int startScan(MultiValueMap<String, MultipartFile> multipartFileMultiValueMap,
+                         Map<String, String[]> parameterMap) {
+
+        int responseStatus = -1;
+        CloseableHttpClient client = null;
+
+        try {
+            URI uri = (new URIBuilder())
```

Comment:
This URL building can be done inside the configuration class itself. That class could contain a getter that returns the required URI.